src/sentry/notifications/helpers.py

```diff
@@ -715,6 +717,20 @@ def get_recipient_from_team_or_user(user_id: int | None, team_id: int | None) ->
     return recipient
 
 
+def team_is_valid_recipient(team: Team | RpcActor, provider: ExternalProviderEnum) -> bool:
+    from sentry.models.integrations.external_actor import ExternalActor
+
+    if provider.value not in TEAM_NOTIFICATION_PROVIDERS:
+        return False
+    linked_integration = ExternalActor.objects.filter(
+        team_id=team.id,
+        provider=EXTERNAL_PROVIDERS_REVERSE[provider].value,
```

Comment:
@isabellaenriquez we should make this work for any external actor for a supported integration. See `PERSONAL_NOTIFICATION_PROVIDERS`.

Reply:
could you clarify what you mean by that? this line only runs when we've confirmed the team provider is a supported one